docs/source/simulator_support.rst

```diff
@@ -223,6 +223,25 @@ In order to use this simulator, set :make:var:`SIM` to ``questa``:
 
     make SIM=questa
 
+Cocotb implements two flows for Questa.
+The most suitable flow is chosen based on the Questa version being used.
+
+The newer **QIS/Qrun flow** uses the Questa Information System (QIS) together with the ``qrun`` command.
+One of the most visible user-facing benefits of the ``qrun`` flow is the ability to automatically order VHDL sources.
+The use of the QIS should reduce the overhead from accessing design internals at runtime and mandates the use of Visualizer as GUI.
+
+The QIS/qrun flow is chosen automatically if Questa 2023.1 or newer is detected.
+Users can explicitly use the QIS/Qrun flow with :make:var:`SIM=questa-qisqrun`.
```

Comment:
Will this cause an issue where the user starts with an old Questa or selected compat and the arguments and all that not working when it magically decides to use the new flow?

Reply:
I added a note to the docs to suggest to people to choose one flow explicitly if they pass simulator arguments.